<?xml version="1.0"?>

<launch>

  <!-- Start Gazebo with field/cow world -->
  <include file="$(find uav_id)/launch/start_world.launch">
  	<arg name="world" value="$(find uav_id)/world/test.world"/>
  	<!-- Start Gazebo without visualisation/GUI (still produces simulated image topic) -->
  	<arg name="gui" value="false"/>
  </include>

  <!-- Spawn simulated quadrotor uav -->
  <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor_with_downward_gimbal_cam.launch" />

  <!-- Start rviz visualization with preset config -->
<!--   <node pkg="rviz" type="rviz" name="rviz" args="-d $(find uav_id)/rviz_cfg/will_test.rviz"/> -->

  <!-- Start image processing node on image topic from UAV -->
  <node name="image_processor" pkg="uav_id" type="ImageProcessor.py" output="screen">
  	<remap from="cam_img" to="/downward_cam/camera/image"/>
  	<param name="v3-network-loc" type="str" value="$(find uav_id)/tensorflow/inceptionv3_retrained.pb"/>
  </node>

</launch>
